rpi3.mk

```diff
@@ -57,22 +58,23 @@ ARM_TF_EXPORTS ?= \
 	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
 
 ARM_TF_FLAGS ?= \
-	BL32=$(OPTEE_OS_BIN) \
+	BL32=$(OPTEE_BIN) \
+	BL32_EXTRA1=$(OPTEE_BIN_EXTRA1) \
+	BL32_EXTRA2=$(OPTEE_BIN_EXTRA2) \
+	BL33=$(U-BOOT_BIN) \
 	DEBUG=1 \
 	V=0 \
 	CRASH_REPORTING=1 \
 	LOG_LEVEL=40 \
 	PLAT=rpi3 \
 	SPD=opteed
 
-arm-tf: optee-os
-	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all
-	cd $(ARM_TF_OUT) && \
-	  dd if=/dev/zero of=scratch bs=1c count=131072 && \
-	  cat $(ARM_TF_BIN) scratch > $(ARM_TF_TMP) && \
-	  dd if=$(ARM_TF_TMP) of=$(ARM_TF_HEAD) bs=1c count=131072 && \
-	  cat $(ARM_TF_HEAD) $(OPTEE_OS_PAGER) > $(ARM_TF_BOOT) && \
-	  rm scratch $(ARM_TF_TMP) $(ARM_TF_HEAD)
+arm-tf: optee-os u-boot
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip
+	@cp $(ARM_TF_OUT)/bl1.bin $(ARM_TF_OUT)/bl1.pad.bin
+	@truncate --size=65536 $(ARM_TF_OUT)/bl1.pad.bin
```

Comment:
This is now 131072 instead of 65536.

Reply:
Actually, take a look at this PR: https://github.com/ARM-software/arm-trusted-firmware/pull/1478 Maybe we can get rid of this completely.